### PR TITLE
`BCOTrainer` conversational dataset support

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Part of #2071

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/bco_trainer.mdx [hunk 1]
```diff
@@ -6,49 +6,8 @@ For a full example have a look at  [`examples/scripts/bco.py`].
...
-        False,
-        False,
-    ],
-}
-```
-
-where the `prompt` contains the context inputs, `completion` contains the corresponding responses and `label` contains the corresponding flag that indicates if the generated completion is desired (`True`) or undesired (`False`).
-A prompt can have multiple responses and this is reflected in the entries being repeated in the dictionary's value arrays. It is required that the dataset contains at least one desirable and one undesirable completion.
-
+The [`BCOTrainer`] requires a [unpaired preference dataset](dataset_formats#unpaired-preference).
```
Comment (lewtun) [inline docs/source/bco_trainer.mdx:9]: turbo nit:
```suggestion
The [`BCOTrainer`] requires an [unpaired preference dataset](dataset_formats#unpaired-preference).
```

File: examples/scripts/bco.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 # Full training:
 python examples/scripts/bco.py \
     --model_name_or_path=nnheui/stablelm-2-1_6b-sft-full \
+    --trust_remote_code \
+    --dataset_name trl-lib/ultrafeedback-gpt-3.5-turbo-helpfulness \
```
Comment (lewtun) [inline examples/scripts/bco.py:22]: I wonder if we can switch this to an unpaired version of `trl-lib/ultrafeedback_binarized`? That way we have two simple datasets that should "just" work

File: examples/scripts/bco.py [hunk 3]
```diff
@@ -18,6 +18,8 @@
 # Full training:
 python examples/scripts/bco.py \
     --model_name_or_path=nnheui/stablelm-2-1_6b-sft-full \
```
Comment (lewtun) [inline examples/scripts/bco.py:20]: Maybe we can use `Qwen2.5-0.5B-Instruct` as a demo model?

Comment (qgallouedec) [reply examples/scripts/bco.py:20]:
```suggestion
    --model_name_or_path Qwen/Qwen2.5-0.5B-Instruct \
```

File: examples/scripts/bco.py (hunk 2 above)
Comment (qgallouedec) [reply examples/scripts/bco.py:22]: Good idea. How `trl-lib/ultrafeedback_binarized` is generated? Which "aspect" (helpfulness, ...) is used?

**Review by kashif — APPROVED** [review]

File: examples/scripts/bco.py (hunk 2 above)
Comment (qgallouedec) [reply examples/scripts/bco.py:22]: I merge the PR, but we can work on it in a following PR

**Pull request merged**